repo/fsrepo/fsrepo.go

```diff
@@ -387,6 +389,42 @@ func (r *FSRepo) SetAPIAddr(addr ma.Multiaddr) error {
 	return err
 }
 
+// SetGatewayAddr writes the Gateway Addr to the /gateway file.
+func (r *FSRepo) SetGatewayAddr(addr net.Addr) error {
+	// Create a temp file to write the address, so that we don't leave empty file when the
+	// program crashes after creating the file.
+	tmpPath := filepath.Join(r.path, "."+gatewayFile+".tmp")
+	f, err := os.Create(tmpPath)
+	if err != nil {
+		return err
+	}
+	var good bool
+	// Silently remove as worst last case with defers.
+	defer func() {
+	  if !good { os.Remove(tmpPath) }
+	}()
+	defer f.Close()
+
+	if _, err := fmt.Fprintf(f, "http://%s", addr.String()); err != nil {
+		return err
+	}
+	if err := f.Close(); err != nil {
+		return err
+	}
+
+	// Atomically rename the temp file to the correct file name.
+	err = os.Rename(tmpPath, filepath.Join(r.path, gatewayFile))
+	good = err == nil
+	if good {
+		return nil
+	}
+	// Remove the temp file when rename return error
+	if err1 := os.Remove(tmpPath); err1 != nil {
+		return fmt.Errorf("File Rename error: %w, File remove error: %s", err.Error(), err1.Error())
```

Comment:
```suggestion
		return fmt.Errorf("File Rename error: %w, File remove error: %s", err, err1.Error())
```
mb, also it need `go fmt ./...`